### DISCUSSION TITLE
Is it possible to delete Applications created with git generator by just deleting the dir on Github?

### DISCUSSION THREAD

**Discussion by mcoklica1** [argoproj/argo-cd#14720] [Q&A]
Hello!

TL;DR the title itself, because all I see on the web is how to delete applicationset and keep your application(s)... we don't want that, since we have a slightly different scenario.

So we have an `applicationset` defined like this, using git directory generator:
```
apiVersion: argoproj.io/v1alpha1
kind: ApplicationSet
metadata:
  name: pr-heavy
  labels:
    name: pr-heavy
spec:
  generators:
  - git:
      repoURL: git@github.com:company/deployment.git
      revision: HEAD
      directories:
      - path: configs/pr/heavy/*
  template:
    metadata:
      name: '{{path.basename}}'
    spec:
      project: '{{path.basename}}'
      source:
        repoURL: git@github.com:company/deployment.git
        targetRevision: HEAD
        path: charts/application
        helm:
          valueFiles:
          - values.yaml
          - '../../configs/pr/heavy/{{path.basename}}/values.yaml'
      destination:
        server: https://kubernetes.default.svc
        namespace: '{{path.basename}}'
      syncPolicy:
        automated:
          prune: false
          selfHeal: false
        syncOptions:
        - CreateNamespace=true
```
Why? We wanted to create a new release (application) every time somebody from dev team creates PR. Our Jenkins will create a folder inside `/deployment/configs/pr/heavy` and generate `values.yaml` file specific to such PR. When the PR is closed/merged, we just told Jenkins to delete the folder with it's contents. Just for the context, the PRs are created for/in another repo (let's call it "application" repo) and inside the "deployment" repo (the one mentioned above) we only have charts and configs for deployment of application master branch and PRs. Our requirements are:
1. Jenkins should not directly invoke commands from Helm or kubectl (only ArgoCD can do that).
2. Applicationset, once defined and applied, should not be deleted... only applications generated from that applicationset should be deleted when PR is closed.

Ok, so far so good, the applicationset creates an application every time a new folder with `values.yaml` inside it comes up in `/deployment/configs/pr/heavy`. But when we delete the folder (meaning: dev closes PR and jenkins executed the operation of deleting the specific dir inside, for example `/deployment/configs/pr/heavy/pr-123` if PR code/ID is 123), our application representing that PR is stuck with Sync Status Unknow and error:
`rpc error: code = Unknown desc = Manifest generation error (cached): ... failed exit status 1: Error: open .configs/pr/heavy/pr-5120/values.yaml: no such file or directory` (I replaced the helm template command with ...).

So again, the question in the title... is this doable? If yes, what we need to do to get our application(s) deleted?
If no, what are our options?

Thanks!

**Comment by crenshaw-dev** [comment]
By default, deleting the directory should cause the Application to be deleted (though that behavior is [configurable](https://argo-cd.readthedocs.io/en/latest/operator-manual/applicationset/Controlling-Resource-Modification/)).

The fact that the app goes out of sync and has a manifest generator error makes sense. The referenced directory is gone. But you should never see the error, because the app should be gone.

Is the ApplicationSet controller logging any errors?

**Comment by mcoklica1** [comment]
Hi @crenshaw-dev,

we have `applicationsetcontroller.policy:create-only` applied in our argocd-applicationset-controller.

there are no level=warn or level=error logs inside our argocd-applicationset-controller. Last log about PR that we tried to delete was around the same time when we terminated the current (in that moment) sync process, because we manually hit/press "Delete" button from the application UI (which wasn't applied immediately, because sync operation was "stuck" the moment we got that `rpc error: code = Unknown desc = Manifest generation error...`).

When it comes to logs about PR (in this case it's called pr-5120), the last message mentioning the PR is this:
`time="2023-07-25T16:34:00Z" level=debug msg="requeue: false caused by application pr-5120\n"`
Before deleting the folder and after deleting the folder in GitHub, majority of messages that mention that PR are like that. Only 4 messages `requeue: true caused by application pr-5120\n"` occurred after deleting the folder in GitHub. Of course, before deleting the folder in GitHub we had messages that start like this: `time="2023-07-25T16:12:15Z" level=debug msg="apps from generator: [{TypeMeta:{Kind: APIVersion:} ObjectMeta:{Name:pr-5120...`.

**Comment by mcoklica1** [comment]
Just a sanity-check:
Does it matter if prune is set to true or false for this scenario? We currently have `prune: false` in our applicationset.

**Comment by mcoklica1** [comment]
Fixed!

> we have applicationsetcontroller.policy:create-only applied in our argocd-applicationset-controller.

Switched to `applicationsetcontroller.policy: create-delete` and deletion was done by ArgoCD with no issues.

Thank you!
